assets/javascripts/discourse/widgets/discourse-reactions-actions.js.es6

```diff
@@ -250,23 +250,20 @@ export default createWidget("discourse-reactions-actions", {
               this.dropUserReaction();
               this.attrs.post.reaction_users_count -= 1;
               this.attrs.post.current_user_used_main_reaction = false;
-              this.setCurrentUserReaction();
+              this.setCurrentUserReaction(null);
               later(() => {
                 dropReaction(postContainer, params.reaction, () => {
-                  return CustomReaction.toggle(
-                    params.postId,
-                    params.reaction
-                  ).then(value => {
-                    resolve();
+                  return CustomReaction.toggle(params.postId, params.reaction)
+                    .then(resolve)
+                    .catch(e => {
+                      bootbox.alert(`${e.jqXHR.status} ${e.errorThrown}`);
 
-                    if (value === undefined) {
                       this.attrs.post.current_user_reaction = current_user_reaction;
```

Comment:
why is it in the catch now ?

Reply:
It was written because, when there's some error in API, we can undo the change we made in the frontend. So, I replaced `then` with `catch` which now works perfectly.

Also, I don't know the reason why but this condition was always true, and due to this, it was causing some issues in UI which was realized now when I testing this condition.

Reply:
please dig a better explanation :)

Reply:
The reason we were using `then` here was, when there's some error in backend we can undo the changes we did from frontend. Eg: we add a 🙂 reaction, but there's 500 error. So in this situation our frontend is updated with 🙂 reaction. To undo this we were using `then` Which caused some issues so i replaced it with `catch` which is used when there are errors in ajax.

Reply:
https://user-images.githubusercontent.com/35112518/106352996-3fbce200-630d-11eb-91a5-e30cf77cc0bc.mov

The issue we were facing in the `then` method. when we drop the reaction from the button, it **re-appeared** due to the condition `if(value === undefined)` was always true, even when the backend doesn't return any error. to solve this I used `catch` which will be called only when there's an error in the backend.